### ISSUE TITLE
Description is unclear for vulnerabilities/stale_feed_data trigger/gate, when fired against an image with a distro/version that doesn't have a feed data source

### ISSUE THREAD

**Issue by nurmi** [anchore/anchore-engine#352]
Currently, when a policy evaluation is performed against an image that reports a distro/version that doesn't match a synced (or existing) OS feed, the output to the user can be confusing - for example, using busybox:latest:

```
vulnerabilities        stale_feed_data                       The vulnerability feed for this image distro is older than MAXAGE (2) days                 warn          
vulnerabilities        vulnerability_data_unavailable        Feed data unavailable, cannot perform CVE scan for distro: busybox:v1.31.1                 warn    
```

first - the 'stale_feed_data' trigger while perhaps true, is misleading since there is no OS feed for the image's distro, and can be understandably interpreted as 'there is a feed, but it is out of date' instead of the more accurate 'there isn't a feed for this distro'.  Suggest that this trigger not fire at all when there is no feed, meaning that the check should only fire if there *is* a feed and it is indeed stale.

second - the vulnerability_data_unavailable detail should be made more explicit to make it clear that this check is only for the OS feed types (as opposed to NVD, other), possibly by introducing additional checks for other feeds and make them more explicit about which type they're referring to.

**Comment by bhearn7** [comment]
@nurmi +1 on this issue. I was confused about these [exact findings](https://repo1.dsop.io/dsop/opensource/thanos/thanos/-/jobs/631305/artifacts/file/ci-artifacts/scan-results/csvs/thanos:v0.16.0-69184-justifications.xlsx) after scanning an image that used a multistage build with quay.io/prometheus/busybox:latest

**Comment by blang9238** [comment]
@nurmi +1 as well, affecting an enterprise customer's end users now.

**Comment by zhill** [comment]
Agreed, I think those triggers need a semantic review and update since we have multiple feeds that are relevant for any given image that are based on content type as much as distro.

**Comment by bhearn7** [comment]
+2, Rancher is seeing this on their images built `FROM scratch`

```
vulnerabilities | stale_feed_data                | The vulnerability feed for this image distro is older than MAXAGE (15) days | stop
vulnerabilities | vulnerability_data_unavailable | Feed data unavailable, cannot perform CVE scan for distro: Unknown:0        | stop
```

**Comment by mohifleur** [comment]
+1 getting the same for `ubuntu:20.10`
```

docker.io/ubuntu:20.10 | vulnerabilities | stale_feed_data                | The vulnerability feed for this image distro is older than MAXAGE (2) days 
docker.io/ubuntu:20.10 | vulnerabilities | vulnerability_data_unavailable | Feed data unavailable, cannot perform CVE scan for distro: ubuntu:20.10

```

**Comment by tyranhenry** [comment]
+1 again on this, affecting another enterprise customer looking at using busybox based images.
